hathor/manager.py

```diff
@@ -1097,7 +1104,7 @@ def sync_v2_step_validations(self, txs: Iterable[BaseTransaction]) -> None:
             try:
                 # XXX: `reject_locked_reward` might not apply, partial validation is only used on sync-v2
                 # TODO: deal with `reject_locked_reward` on sync-v2
-                assert tx.validate_full(reject_locked_reward=True)
+                assert tx.validate_full(reject_locked_reward=False)
```

Comment:
This actually has to be `False`, it's important that there is a "consensus update" (actually just a check) after the round of "step validations" to check and remove transactions that were affected by the reward lock.

The reason this needs to be `False` is because this validations can happen before the chain of blocks is updated enough to affect the reward lock check that is being skipped. This check is `True` in all other cases because in those cases we are either adding transactions in real time or in "timestamp order" (on sync-v1). But on sync-v2, as much as we prioritize blocks, we don't guarantee in practice that the validation order will be "timestamp sorted".

It's important to note that the "min_height" reward lock validation is never disabled, only the "transaction reward lock". The "transaction reward lock" validation only has any practical effect for transactions which have not been confirmed by a block, that's why it's correct to use `True` here, as long as there is the "clean up pass" (which currently only happens on a consensus update.

But to be clear, I'm not sure if there is a consensus update after all cases where the "step validations" happen. There also is not a specific test for this, this is only indirectly and incidentally covered by other tests.